turbopack/crates/turbopack-ecmascript/src/analyzer/graph.rs

```diff
@@ -1333,30 +1380,43 @@ impl VisitAstPath for Analyzer<'_> {
         new_expr: &'ast NewExpr,
         ast_path: &mut AstNodePath<AstParentNodeRef<'r>>,
     ) {
-        // new URL("path", import.meta.url)
         if let box Expr::Ident(ref callee) = &new_expr.callee {
-            if &*callee.sym == "URL" && is_unresolved(callee, self.eval_context.unresolved_mark) {
+            if is_unresolved(callee, self.eval_context.unresolved_mark) {
                 if let Some(args) = &new_expr.args {
-                    if args.len() == 2 {
-                        if let Expr::Member(MemberExpr {
-                            obj:
-                                box Expr::MetaProp(MetaPropExpr {
-                                    kind: MetaPropKind::ImportMeta,
-                                    ..
-                                }),
-                            prop: MemberProp::Ident(prop),
-                            ..
-                        }) = &*args[1].expr
-                        {
-                            if &*prop.sym == "url" {
-                                self.add_effect(Effect::Url {
-                                    input: self.eval_context.eval(&args[0].expr),
-                                    ast_path: as_parent_path(ast_path),
-                                    span: new_expr.span(),
-                                    in_try: is_in_try(ast_path),
-                                });
+                    match &*callee.sym {
+                        // new URL("path", import.meta.url)
+                        "URL" if args.len() == 2 => {
+                            if let Expr::Member(MemberExpr {
+                                obj:
+                                    box Expr::MetaProp(MetaPropExpr {
+                                        kind: MetaPropKind::ImportMeta,
+                                        ..
+                                    }),
+                                prop: MemberProp::Ident(prop),
+                                ..
+                            }) = &*args[1].expr
+                            {
+                                if &*prop.sym == "url" {
+                                    self.add_effect(Effect::Url {
```

Comment:
Same argument for `Effect::Url`. It shouldn't exist. Instead `Effect::Call { ..., new: true }` should be handled during linking if the callee is `WellKnownFunciton::Url`